services/training_service.py

```diff
@@ -17,11 +18,17 @@ def train():
             traceback.print_exc()
             print(exc)
 
-    @staticmethod
-    def populate_training(form):
-        password = form.get('password')
+    def populate_training(self, request):
+        username = request.headers.get('username')
+        password = request.headers.get('password')
+        url = request.form.get('url')
+
+        if username != 'username' or password != 'password':
+            return jsonify({'error': 'Wrong username or password'}), 401
+
+        if not url:
+            return jsonify({'error': 'Url not provided'}), 400
```

Comment:
1- For everything related to the HTTP, I would do it in the routes.py.
everything related to the business logic  I would do in the service
2- Take the password from an environment variable, not hard coded